src/workspace-data/data-table/entity-service/EntityDeleter.js

```diff
@@ -27,11 +27,20 @@ export const EntityDeleter = ({ onDismiss, onSuccess, namespace, name, selectedE
     } catch (error) {
       switch (error.status) {
         case 409:
-          setAdditionalDeletions(_.filter((entity) => entity.entityType !== selectedDataType, await error.json()));
+          setAdditionalDeletions(
+            _.filter(
+              (errorEntity) =>
+                !_.some(
+                  (selectedEntity) => selectedEntity.entityType === errorEntity.entityType && selectedEntity.entityName === errorEntity.entityName,
+                  entitiesToDelete
+                ),
+              await error.json()
+            )
+          );
```

Comment:
Separate the selected entities from the additional entities to ensure the additional entities are deleted first, followed by the selected entities."